app/src/main/kotlin/com/pitchedapps/frost/utils/Prefs.kt

```diff
@@ -153,7 +157,15 @@ object Prefs : KPref() {
 
     var verboseLogging: Boolean by kpref("verbose_logging", false)
 
-    var analytics: Boolean by kpref("analytics", true)
+    var analytics: Boolean by kpref("analytics", false) {
+        if (!BuildConfig.DEBUG) {
+            if (it) {
+                Bugsnag.enableExceptionHandler()
+            } else {
+                Bugsnag.disableExceptionHandler()
```

Comment:
I just checked on my phone, disabling the exception handler isn't enough to disable bugsnag unfortunately (the connection is still open), a restart is required (or just inform the user that the preference will be applied next time the application is started)

Reply:
I will send bugsnag an email later. It may be that I also have to disable stage notifications, which is probably what it sends at first

Reply:
Updates on this? Do they have a way to disable bugsnag after initialization?

Reply:
The initial report is sent on init, so they basically said to late init it. I might end up requiring an app restart since I don't like the idea of init on activity. It also depends on if bugsnag crashes if I fail to init, which I will have to test on my own later

Reply:
Nothing wrong with requiring an app restart in my opinion. It's not the most elegant solution, but with lack of a better option, I'd do it

Reply:
They've provided a better solution today. I will work on it tomorrow

Reply:
If you want you can try this out. I'm reasonably happy with the solution, and given that everything is behind the analytics flag there shouldn't be anything unexpected

Reply:
Are you still calling initBugsnag before the consent screen? Because that's when it sends an initial batch of data about the device and leaves the connection open for breadcrumbs and crash reports.